### PR TITLE
Remove the usage of dataclasses

### PR DESCRIPTION
**Pull request by plule-ansys**
The dataclass pattern looked great for removing the boilerplate of __init__, __eq__ around attributes.

However, in the end there was multiple issues in our context:
 - Does not play nicely with sphinx. As soon as we try to customize an attribute using "field()", it would hide it from the doc (or create errors)
 - Does not allow for private setters, classes and attributes are either writable or read-only. This does not play well in Instance for example, where we want it read-only for the user, but still updated when calling "update()".
 - Does not play well with private attributes like the _stub: the constructor argument itself is "_stub".

This PR switches to a more usual way with private attribute and their setters.

### CODE REVIEW

**Comment by plule-ansys** [comment]
I doubled check: no doc impact (attribute are reordered but that's all)

File: src/ansys/platform/instancemanagement/definition.py [hunk 1]
```diff
@@ -13,44 +12,82 @@
...
-    """
+    @property
+    def product_version(self) -> str:
+        """Version of the product.
 
-    available_service_names: Sequence[str]
-    """List of the available service names.
+        This is a string describing the version.
+        When the product is following the release process for the Ansys unified installation,
+        the version is three numbers, such as "221".
```
Comment (jerome-blanche) [inline src/ansys/platform/instancemanagement/definition.py:51]: suggestion: three digits

File: src/ansys/platform/instancemanagement/instance.py [hunk 2]
```diff
@@ -31,54 +30,84 @@
...
 
-    status_message: str
-    """Status of the instance.
+    @property
+    def ready(self) -> bool:
+        """Whether the instance is ready.
 
-    Human-readable message describing the status of the instance.
-    This field is always filled when the instance is not ready."""
+        If ``True``, the ``services`` field contains the list of entry points
```
Comment (jerome-blanche) [inline src/ansys/platform/instancemanagement/instance.py:65]: suggestion: replace `field` by `property` in the doc strings for the `ready`, `status_message`, and `services` properties.

**Review by jerome-blanche — APPROVED** [review]

File: src/ansys/platform/instancemanagement/definition.py (hunk 1 above)
Comment (plule-ansys) [reply src/ansys/platform/instancemanagement/definition.py:51]: Sure, done thanks

**Pull request merged**